django/contrib/staticfiles/storage.py

```diff
@@ -84,12 +84,12 @@ def file_hash(self, name, content=None):
             md5.update(chunk)
         return md5.hexdigest()[:12]
 
-    def hashed_name(self, name, content=None):
+    def hashed_name(self, name, content=None, accept_fragments=False):
         parsed_name = urlsplit(unquote(name))
         clean_name = parsed_name.path.strip()
         opened = False
         if content is None:
-            if not self.exists(clean_name):
+            if not (accept_fragments or self.exists(clean_name)):
```

Comment:
This has two undesireable side effects:

1. It generates file names without a hash, defeating the purpose of `HashedFilesMixin`. This will result in incorrect caching while `HashedFilesMixin` precisely exists to provide cache invalidation when static files change.

2. It lets errors slip through silently when a file doesn't exist. This may be considered a feature or a bug, depending on the developer's expectations. (Some developers consider that a missing CSS file or image isn't an issue. I don't know if they would also advocate not raising an error when a Python module isn't found ;-) I consider that it should raise an exception.)